presto-rcfile/src/main/java/io/prestosql/rcfile/binary/TimestampEncoding.java

```diff
@@ -36,13 +41,17 @@
 public class TimestampEncoding
         implements BinaryColumnEncoding
 {
-    private final Type type;
+    private final TimestampType type;
     private final DateTimeZone timeZone;
+    private final PrestoTimestampEncoder<?> prestoTimestampEncoder;
 
     public TimestampEncoding(Type type, DateTimeZone timeZone)
     {
-        this.type = requireNonNull(type, "type is null");
+        requireNonNull(type, "type is null");
+        verify(type instanceof TimestampType, "type is not a TimestampType");
+        this.type = (TimestampType) type;
```

Comment:
The cast should be done where we check the type (in `io.prestosql.rcfile.RcFileEncoding#getEncoding`)